api/resource.yaml

```diff
@@ -0,0 +1,2200 @@
+openapi: 3.0.3
+info:
+  title: Resource Server Management API
+  version: "1.0"
+  description: |
+    This API is used to manage resource servers, resources, and actions for permission-based access control.
+    Resources can be nested hierarchically, and permissions are derived from the name-based path structure.
+    Resource and Action names are immutable after creation and used directly for permission derivation.
```

Comment:
Do we need to make Resource/ Action name immutable? handle is the one being immutable right?